### PR TITLE
[NEW] URL mappings for servlet modules

### PR DESCRIPTION
**Pull request by monu-k2io**
Feature for getting url mappings for servlet 2.4, 5.0 and 6.0 
Feature for getting jsp(s)

### CODE REVIEW

File: instrumentation-security/servlet-2.4/src/main/java/javax/servlet/ServletContainerInitializer_Instrumentation.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public void onStartup(Set<Class<?>> c, ServletContext ctx) throws ServletException {
+        try {
+            Weaver.callOriginal();
+        } finally {
+            postProcessing(ctx);
+        }
+    }
+
+    private static void postProcessing(ServletContext context) {
+        try {
```
Comment (lovesh-ap) [inline instrumentation-security/servlet-2.4/src/main/java/javax/servlet/ServletContainerInitializer_Instrumentation.java:21]: No need of try catch as inner method already contains it

File: instrumentation-security/servlet-5.0/src/main/java/com/nr/instrumentation/security/servlet5/HttpServletHelper.java [hunk 2]
```diff
@@ -105,4 +112,33 @@ public static void releaseServletLock() {
...
+                    URLMappingsHelper.addApplicationURLMapping(new ApplicationURLMapping(WILDCARD, s));
+                }
+            }
+        } catch (Exception ignored){
+        }
+    }
+
+    public static void getJSPMappings(ServletContext servletContext, String dir) {
+        try {
+            if(dir.endsWith(SEPARATOR)){
```
Comment (lovesh-ap) [inline instrumentation-security/servlet-5.0/src/main/java/com/nr/instrumentation/security/servlet5/HttpServletHelper.java:132]: Will this JSP mappings work on windows platform as well? as the SEPARATOR is set to "/"

Comment (IshikaDawda) [reply instrumentation-security/servlet-5.0/src/main/java/com/nr/instrumentation/security/servlet5/HttpServletHelper.java:132]: Yes, this is working on windows platform as well.

**Review by lovesh-ap — APPROVED** [review]

File: instrumentation-security/servlet-2.4/src/main/java/javax/servlet/ServletContainerInitializer_Instrumentation.java (hunk 1 above)
Comment (IshikaDawda) [reply instrumentation-security/servlet-2.4/src/main/java/javax/servlet/ServletContainerInitializer_Instrumentation.java]: Fixed this in commit [8652436](https://github.com/newrelic/csec-java-agent/pull/52/commits/86524360e0e9d71dc5696d7da747931d01b310df)

**Pull request merged**